### PR TITLE
Add extensions version syntax support docs

### PR DESCRIPTION
**Pull request by gtsiolis**
This will add a small note about extensions version syntax support the `.gitpod.yml` reference docs. 📖 

See also [relevant discussion](https://gitpod.slack.com/archives/C01LDA27WJC/p1625155283215000) (internal). Cc @akosyakov @jankeromnes @mikenikles 

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/717"></a>

### CODE REVIEW

File: src/routes/docs/references/gitpod-yml.md [hunk 1]
```diff
@@ -372,13 +372,15 @@ Please note, Gitpod uses the [Open VSX registry](https://open-vsx.org/) to find
 | -------- | --------- |
 | `object` | `<empty>` |
 
+By default, extensions will use the latest available version unless you use a specific version number. The version number must use semantic versioning rules.
```
Comment (gtsiolis) [inline src/routes/docs/references/gitpod-yml.md:375]: Adding below one more thing about proprietrary extensions.

```suggestion
By default, extensions will use the latest available version unless you use a specific version number. The version number must use semantic versioning rules. If you are interested in importing an extension that is not published on the Open VSX registry you can directly use the full URL.
```

File: src/routes/docs/references/gitpod-yml.md [hunk 2]
```diff
@@ -372,13 +372,15 @@ Please note, Gitpod uses the [Open VSX registry](https://open-vsx.org/) to find
...
+
 **Example**
 
 ```yaml
 vscode:
   extensions:
     - svelte.svelte-vscode
-    - bradlc.vscode-tailwindcss
+    - bradlc.vscode-tailwindcss@0.6.11
```
Comment (gtsiolis) [inline src/routes/docs/references/gitpod-yml.md:383]:
```suggestion
    - bradlc.vscode-tailwindcss@0.6.11
    - https://example.come/abc/releases/extension-0.26.0.vsix 
```

**Review by gtsiolis — COMMENTED** [review]
Added one more point about consuming proprietrary extensions. Thanks for the ping @akosyakov!

**Review by akosyakov — APPROVED** [review]
Thank you!

**Review by mikenikles — APPROVED** [review]
Thank you George ❤️!

**Comment by gtsiolis** [comment]
Thanks for taking care of this, @mikenikles! 🏀

**Pull request merged**